include/sys/mem_manage.h

```diff
@@ -14,14 +16,26 @@
  * Caching mode definitions. These are mutually exclusive.
  */
 
-/** No caching. Most drivers want this. */
-#define K_MEM_CACHE_NONE	2
+/** Full write-back caching. Any RAM mapped wants this. */
+#define K_MEM_CACHE_WB		0
 
 /** Write-through caching. Used by certain drivers. */
 #define K_MEM_CACHE_WT		1
 
-/** Full write-back caching. Any RAM mapped wants this. */
-#define K_MEM_CACHE_WB		0
+/** No caching. Most drivers want this. */
+#define K_MEM_CACHE_NONE	2
+
+/*
+ * ARM64 Specific flags: device memory with nGnRnE.
+ * For ARM64, K_MEM_CACHE_NONE is nGnRnE.
+ */
+#define K_MEM_ARM_DEVICE_nGnRnE	K_MEM_CACHE_NONE
+
+/** ARM64 Specific flags: device memory with nGnRE */
+#define K_MEM_ARM_DEVICE_nGnRE	3
+
+/** ARM64 Specific flags: device memory with GRE */
+#define K_MEM_ARM_DEVICE_GRE	4
```

Comment:
Uhm, not a fan of this because we are polluting this header file with arch-specific stuff. So either we move this to an ARM-specific header to include directly, or we ifdef include an ARM-specific header from this file.

Reply:
Agreed, `include/sys/mem_manage.h` seems a system-wide header. What about putting these into `include/arch/arm64/mm.h` which could be beneficial to both mmu and mpu?

Reply:
Hi, @povergoing and @carlocaione, thanks for your comments, these ARM-specific are also flags used by z_phys_map(), so if we put them in include/arch/arm64/mm.h, it will be difficult to keep these definition to be not conflicted with K_MEM_* definitions in include/sys/mem_manage.h in case of there will be some update in include/sys/mem_manage.h, any better solution?

So how about add a if deffined(CONFIG_ARM_MMU) || defined(CONFIG_ARM_MPU) to limit these ARM specific definitions?

Thanks.

Reply:
that's why I suggested to have in this file something like:
```
#if defined(CONFIG_ARM_MMU)
#include <something_arm_specific.h>
#endif
```

Reply:
Updated, please review again, thanks.

Reply:
It seems there are some build errors when we include` arch/arm64/mm.h `in `include/sys/mem_manage.h`, so how about introduce a new header file to include these definitions, such as ` arch/arm64/arm_mem.h `?

zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -c /__w/zephyr/zephyr/arch/arm/core/offsets/offsets.c
In file included from ../include/arch/arm64/mm.h:17,
                 from ../include/sys/mem_manage.h:13,
                 from ../include/sys/device_mmio.h:43,
                 from ../include/device.h:31,
                 from ../kernel/include/kernel_offsets.h:6,
                 from /__w/zephyr/zephyr/arch/arm/core/offsets/offsets_aarch32.c:30,
                 from /__w/zephyr/zephyr/arch/arm/core/offsets/offsets.c:9:
../include/arch/arm64/cortex_r/arm_mpu.h:143: warning: "REGION_RAM_ATTR" redefined
  143 | #define REGION_RAM_ATTR            \
      | 
In file included from ../include/arch/arm/aarch32/arch.h:197,
                 from ../include/arch/cpu.h:19,
                 from ../include/kernel_includes.h:33,
                 from ../include/kernel.h:17,
                 from /__w/zephyr/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /__w/zephyr/zephyr/arch/arm/core/offsets/offsets.c:9:
../include/arch/arm/aarch32/mpu/nxp_mpu.h:102: note: this is the location of the previous definition
  102 | #define REGION_RAM_ATTR   {((MPU_REGION_SU_RW) | \
      |

Reply:
fine to me as long as it compiles :)

Reply:
Done, all compile pass, please review, thanks.

Reply:
LGTM, apologize for the improper suggestion.